core-web/libs/edit-content/src/lib/store/features/lock.feature.ts

```diff
@@ -0,0 +1,216 @@
+                const currentUser = store.currentUser();
+                const isLockedByCurrentUser = currentUser.userId === lockedBy.userId;
+
+                const userDisplay = isLockedByCurrentUser
+                    ? dotMessageService.get('edit.content.locked.by.you')
+                    : lockedBy.firstName + ' ' + lockedBy.lastName;
+
+                return dotMessageService.get('edit.content.locked.toolbar.message', userDisplay);
+            })
+        })),
+
+        withMethods(
+            (
+                store,
+                dotContentletService = inject(DotContentletService),
+                dotHttpErrorManagerService = inject(DotHttpErrorManagerService)
+            ) => ({
+                /**
+                 * Locks the current content
+                 *
+                 * Makes an API call to lock the content and updates the store with the returned contentlet
+                 * Clears any previous lock errors before attempting to lock
+                 */
+                lockContent: rxMethod<void>(
+                    pipe(
+                        tap(() =>
+                            patchState(store, {
+                                lockError: null
+                            })
+                        ),
+                        switchMap(() =>
+                            dotContentletService.lockContent(store.contentlet()?.inode).pipe(
+                                tapResponse({
+                                    next: (contentlet: DotCMSContentlet) => {
+                                        patchState(store, {
+                                            contentlet
+                                        });
+                                    },
+                                    error: (error: HttpErrorResponse) => {
+                                        dotHttpErrorManagerService.handle(error);
+                                        patchState(store, {
+                                            lockError: error.message
+                                        });
+                                    }
+                                })
+                            )
+                        )
+                    )
+                ),
+
+                /**
+                 * Unlocks the current content
+                 *
+                 * Makes an API call to unlock the content and updates the store with the returned contentlet
+                 * Clears any previous lock errors before attempting to unlock
+                 */
+                unlockContent: rxMethod<void>(
+                    pipe(
+                        tap(() =>
+                            patchState(store, {
+                                lockError: null
+                            })
+                        ),
+                        switchMap(() =>
+                            dotContentletService.unlockContent(store.contentlet()?.inode).pipe(
```

Comment:
same here, exist a visible loading state?

Reply:
same here, not really a loading state.. this is a visual change in a input switch, and when locked the form will be disabled.